lib/galaxy/schema/schema.py

```diff
@@ -0,0 +1,359 @@
+    )
+    links: List[Hyperlink] = Field(
+        ...,
+        title="Links",
+        description="The collection of link details for this Display Application.",
+    )
+
+
+class Visualization(BaseModel):  # TODO annotate this model
+    class Config:
+        extra = Extra.allow  # Allow any fields temporarily until the model is annotated
+
+
+class HistoryItemBase(BaseModel):
+    id: EncodedDatabaseIdField = Field(
+        ...,
+        title="ID",
+        description="The encoded ID of the item.",
+    )
+    name: str = Field(
+        ...,
+        title="Name",
+        description="The name of the dataset.",
+    )
+    history_id: EncodedDatabaseIdField = Field(
+        ...,
+        title="History ID",
+        description="The encoded ID of the history containing this item.",
+    )
+    hid: int = Field(
+        ...,
+        title="HID",
+        description="The index position of this item in the History.",
+    )
+    history_content_type: HistoryContentType = Field(
+        ...,
+        title="Content Type",
+        description="The type of this item.",
+    )
+    deleted: bool = Field(
+        ...,
+        title="Deleted",
+        description="Whether this item is marked as deleted.",
+    )
+    visible: bool = Field(
+        ...,
+        title="Visible",
+        description="Whether this item is visible or hidden to the user by default.",
+    )
+
+
+class HistoryItemSummary(HistoryItemBase):
+    type_id: str = Field(
+        ...,
+        title="Type - ID",
+        description="The type and the encoded ID of this item. Used for caching.",
+        example="dataset-616e371b2cc6c62e",
+    )
+    type: str = Field(
+        ...,
+        title="Type",
+        description="The type of this item.",
+    )
+    create_time: datetime = Field(
+        ...,
+        title="Create Time",
+        description="The time and date this item was created.",
+    )
+    update_time: datetime = Field(
+        ...,
+        title="Update Time",
+        description="The last time and date this item was updated.",
+    )
+    url: AnyUrl = UrlField
+    tags: TagCollection
+
+
+class HDASummary(HistoryItemSummary):
+    dataset_id: EncodedDatabaseIdField = Field(
+        ...,
+        title="Dataset ID",
+        description="The encoded ID of the dataset associated with this item.",
+    )
+    state: Dataset.states = DatasetStateField
+    extension: str = Field(
+        ...,
+        title="Extension",
+        description="The extension of the dataset.",
+        example="txt",
+    )
+    purged: bool = Field(
+        ...,
+        title="Purged",
+        description="Whether this dataset has been removed from disk.",
+    )
+
+
+class HDAInaccessible(HistoryItemBase):
+    accessible: bool = AccessibleField
+    state: Dataset.states = DatasetStateField
+
+
+class HDADetailed(HDASummary):
+    model_class: str = ModelClassField(HDA_MODEL_CLASS_NAME)
```

Comment:
It's really awesome to see this all in one place, it makes the duplicated and/or expensive fields very visible.